addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/BigRedirectsScannerTest.java

```diff
@@ -0,0 +1,123 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+import net.htmlparser.jericho.Source;
+import org.apache.commons.httpclient.URI;
+import org.junit.Before;
+import org.junit.Test;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+
+public class BigRedirectsScannerTest extends PassiveScannerTest<BigRedirectsScanner> {
+    private static final String URI = "http://example.com";
+
+    protected HttpMessage msg;
+
+    @Override
+    protected BigRedirectsScanner createScanner() {
+        return new BigRedirectsScanner();
+    }
+
+    @Test
+    public void GivenRedirectWithSmallBodyThenItRaisesNoAlert() {
+        msg.getResponseHeader().setStatusCode(300);
+        msg.getResponseHeader().setHeader("Location", URI);
+        msg.setResponseBody(new byte[allowableBodySize()]);
+
+        rule.scanHttpResponseReceive(msg, -1, new Source(""));
+
+        assertThat(alertsRaised.size(), is(0));
+    }
+
+    @Test
+    public void GivenRedirectHeadersWithLargeBodyThenAlertRaised() {
+        msg.getResponseHeader().setStatusCode(300);
+        msg.getResponseHeader().setHeader("Location", URI);
+        msg.setResponseBody(new byte[allowableBodySize() + 1]);
+
+        rule.scanHttpResponseReceive(msg, -1, new Source(""));
+
+        assertThat(alertsRaised.size(), is(1));
+        assertBigRedirectAlertAttributes(alertsRaised.get(0));
+    }
+
+    @Test
+    public void GivenNotModifiedStatusCodeWithLargeBodyThenNoAlertRaised() {
+        msg.getResponseHeader().setStatusCode(304);
+        msg.getResponseHeader().setHeader("Location", URI);
+        msg.setResponseBody(new byte[allowableBodySize() + 1]);
+
+        rule.scanHttpResponseReceive(msg, -1, new Source(""));
+
+        assertThat(alertsRaised.size(), is(0));
+    }
+
+    @Test
+    public void GivenRedirectStatusCodeWithoutLocationHeaderThenNoAlertRaised() {
+        msg.getResponseHeader().setStatusCode(300);
+        msg.setResponseBody(new byte[allowableBodySize() + 1]);
+
+        rule.scanHttpResponseReceive(msg, -1, new Source(""));
+
+        assertThat(alertsRaised.size(), is(0));
+    }
+
+    @Before
+    public void CreateHttpMessage() throws IOException {
```

Comment:
This method seems unused.

Reply:
It's part of the set up, with the `@Before` tag, to create a common `HTTPMessage`.

Reply:
Oopps, totally overlooked the `@Before` annotation.

As with all the other methods it should use standard camelCase (leading lower).

Reply:
Thanks! I don't write Java very frequently.  How about the other `public` methods? Should they begin with a lowercase `given`?

Reply:
The comments (Given, When, Then) are fine, it was just the method names that I was concerned about.

Reply:
Awesome, do you have any other feedback for this PR?

Reply:
Ya sorry I'm not being clear. You were right two comments ago, I misunderstood what you were asking.
`GivenRedirectWithSmallBodyThenItRaisesNoAlert` > `givenRedirectWithSmallBodyThenItRaisesNoAlert` and similar for everything.

Again sorry for not being clear.

Reply:
Nor a problem. 🤦‍♂